lib/launcher/src/Cardano/Launcher.hs

```diff
@@ -119,14 +119,22 @@ launch cmds = do
             "Unreachable. Supervising threads should never finish. " <>
             "They should stay running or throw @ProcessHasExited@."
 
+data FileExists = FileExists
+    deriving Show
+
+instance Exception FileExists
+
 -- | Initialize a state directory to store blockchain data such as blocks or
 -- the wallet database.
-setupStateDir :: (Text -> IO ()) -> (FilePath -> IO ()) -> FilePath -> IO ()
-setupStateDir logT withDir dir = do
+setupStateDir :: (Text -> IO ()) -> (FilePath -> IO ()) -> FilePath -> IO (Either FileExists ())
+setupStateDir logT withDir dir = try $ do
```

Comment:
Actually, I think this function could return `IO ()` and call `exitFailure` directly. In a small program like this it's fine.

Reply:
That was my first thought also but I would end with giving the message with `putStrLn` which I didn't like (I thought that giving this message with `putErrLn` from `Cardano.CLI` would be nicer but couldn't import it here because of cycle)

Reply:
@piotr-iohk Note that, to cope with import cycles here, the function receives a logger as argument `Text -> IO ()`. Without much effort, we can make this `Severity -> Text -> IO ()`. Another option could be to move this function into `Cardano/CLI`.

Reply:
I get it, there's several options... :) which one would be preferred over the existing one?

Reply:
I think moving code will be the clearer path in this case.